### DISCUSSION TITLE
路由path动态参数配置问题求助

### DISCUSSION THREAD

**Discussion by Wei0809** [umijs/umi#11848] [Q&A]
按照文档这样配置的路由
```
{
          name: 'Demand EP',
          path: '/demand/demand-ep/:demandId',
          component: './Demand/DemandEP',
        }
```
在菜单中直接点击会出现这样的情况：
![image](https://github.com/umijs/umi/assets/43309547/c855855f-c369-4c20-90ca-61e5228cf38d)
请问是配置问题吗

**Comment by fz6m** [comment]
如有问题，需要给一个最小复现。

**Reply by Wei0809** [reply]
我搞了个demo，就是在点击testPageA时，菜单会直接把我配置的动态id `/:id` 给拼在链接上 [codeSandBox](https://codesandbox.io/p/github/Wei0809/umi-Example/master?import=true&embed=1&file=%2F.umirc.ts&layout=%257B%2522sidebarPanel%2522%253A%2522GIT%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cloqx4tky00vq3b6hjmdqvrpt%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cloqx4tky00vn3b6hwsbp7nf7%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cloqx4tky00vp3b6hp16p65h9%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cloqx4tky00vo3b6hpwez4lfp%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cloqx4tky00vn3b6hwsbp7nf7%2522%253A%257B%2522id%2522%253A%2522cloqx4tky00vn3b6hwsbp7nf7%2522%252C%2522activeTabId%2522%253A%2522cloqxanue02wt3b6h5zgb239a%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fpages%252FTestPage%252FTestPageA%252Findex.tsx%2522%252C%2522id%2522%253A%2522cloqx84ja017c3b6hno1w4lev%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.umirc.ts%2522%252C%2522id%2522%253A%2522cloqxanue02wt3b6h5zgb239a%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.npmrc%2522%252C%2522id%2522%253A%2522cloqxekfp03oz3b6hgdjgw0k6%2522%252C%2522mode%2522%253A%2522temporary%2522%257D%255D%257D%252C%2522cloqx4tky00vo3b6hpwez4lfp%2522%253A%257B%2522id%2522%253A%2522cloqx4tky00vo3b6hpwez4lfp%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A8000%252C%2522id%2522%253A%2522cloqx6d23013r3b6hqdktukm2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252Fhome%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cloqx6d23013r3b6hqdktukm2%2522%257D%252C%2522cloqx4tky00vp3b6hp16p65h9%2522%253A%257B%2522id%2522%253A%2522cloqx4tky00vp3b6hp16p65h9%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cloqwz21x00043b6hj3qzafn9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cloqwz2dw001jeei26dhn641i%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522cloqwz34i007d3b6h42v5z2t5%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522CSB_RUN_OUTSIDE_CONTAINER%253D1%2520devcontainer%2520templates%2520apply%2520--template-id%2520%255C%2522ghcr.io%252Fdevcontainers%252Ftemplates%252Ftypescript-node%255C%2522%2520--template-args%2520%27%257B%257D%27%2520--features%2520%27%255B%255D%27%2522%252C%2522id%2522%253A%2522cloqx0ypl00bf3b6hchheo7dv%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%252C%2522activeTabId%2522%253A%2522cloqx0ypl00bf3b6hchheo7dv%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by fz6m** [reply]
你配置了什么 path 就会有这个菜单，实际跳转到这个 path ，如果希望保留这个动态页面，但不在菜单上显示，可以去掉 `name` 。

**Reply by Wei0809** [reply]
就是我想保留这个菜单，同时也希望配置动态参数的方案是不行的吗？主要是希望在菜单中跳转时path里面不会跟配置的动态参数字段，这个能否作为一个feature呢？

**Reply by fz6m** [reply]
如果你想要一个一级菜单里面只有一个二级菜单的话，就让 `routes` 只有一个对象，之后继续嵌套更细致的 `routes` 配置实现即可（如果不想嵌套，可以配成 `*` 去组件里判断路径），以下是一个细致配置的例子：

```ts
  {
      name: 'TestPage',
      path: '/test-page',
      routes: [
        // ↓ 这里只有一个对象，所以二级菜单只有一个，点击后会跳转到 `/test-page`
        {
          name: 'TestPageA',
          path: '/test-page/*',
          routes: [
            // ↓ 这里面的三级菜单，因为没有 name 所以不会渲染，做细致区分即可
            
            // ↓让 `/test-page` 也和 `/test-page/:id` 渲染一样的或者重定向到具体路径，`component` 和 `redirect` 只能二取一。
            {
              path: '',
              component: './TestPage/TestPageA',
              // redirect: '/test-page/1',
            },
            {
              path: 'specified',
              component: './TestPage/TestPageB',
            },
            {
              path: ':id',
              component: './TestPage/TestPageA',
            },
          ]
        },
      ],
    }
```

**Reply by Wei0809** [reply]
大概明白了，感谢，但我觉得当在path里配置 /test-page/:id 时，点击菜单跳转时，路径里不应该有 :id 这个参数 ，而是直接跳转/test-page或者 /test-page/ 😂
